### PR TITLE
Clean up a little usage of get_X_question_filters

### PR DESCRIPTION
**Pull request by kongzii**

### CODE REVIEW

File: prediction_market_agent_tooling/markets/omen/omen_subgraph_handler.py [hunk 1]
```diff
@@ -218,30 +218,45 @@ def _get_fields_for_markets(self, markets_field: FieldPath) -> list[FieldPath]:
...
-        condition_id_in: list[HexBytes] | None = None,
-        id_in: list[str] | None = None,
-        excluded_questions: set[str] | None = None,
-        collateral_token_address_in: tuple[ChecksumAddress, ...] | None = None,
-        category: str | None = None,
+        creator: HexAddress | None,
+        creator_in: t.Sequence[HexAddress] | None,
+        outcomes: list[str],
+        created_after: DatetimeUTC | None,
+        question_opened_before: DatetimeUTC | None,
```
Comment (kongzii) [inline prediction_market_agent_tooling/markets/omen/omen_subgraph_handler.py:225]: Removed `= None` default values, so if we forget to pass the filter somewhere, mypy will warn us.

File: prediction_market_agent_tooling/markets/omen/omen_subgraph_handler.py [hunk 2]
```diff
@@ -218,30 +218,45 @@ def _get_fields_for_markets(self, markets_field: FieldPath) -> list[FieldPath]:
...
+        question_opened_after: DatetimeUTC | None,
```
Comment (kongzii) [inline prediction_market_agent_tooling/markets/omen/omen_subgraph_handler.py:226]: Also prepended the filters that are passed to the Question object with `question_`.

File: prediction_market_agent_tooling/markets/omen/omen_subgraph_handler.py [hunk 3]
```diff
@@ -644,15 +641,21 @@ def get_resolved_bets_with_valid_answer(
...
+        finalized_after: DatetimeUTC | None,
+        with_anwers: bool | None,
+        question_id: HexBytes | None,
+        question_id_in: list[HexBytes] | None,
+        opened_before: t.Optional[DatetimeUTC],
+        opened_after: t.Optional[DatetimeUTC],
+        excluded_titles: set[str] | None,
     ) -> dict[str, t.Any]:
+        """
+        Be aware, both Omen subgraph and Reality subgraph are indexing questions, but their fields are a bit different.
```
Comment (kongzii) [inline prediction_market_agent_tooling/markets/omen/omen_subgraph_handler.py:657]: This is where I was stuck both `realityeth_subgraph` and `trades_subgraph` are indexing questions but the fields are a little different.

Maybe we shouldn't have a single OmenSubgraphHandler class, but per-subgraph classes like `TradesSubgraphHandler`, `RealitySubgraphHandler`, etc. But that's for another PR.

Here I split the wheres-creation into `get_reality_question_filters` used on reality subgraph, and `get_omen_question_filters` used on Omen subgraph.

Comment (kongzii) [reply prediction_market_agent_tooling/markets/omen/omen_subgraph_handler.py:657]: Where possible, I added the missing filters that the other one had.

File: prediction_market_agent_tooling/markets/omen/data_models.py [hunk 4]
```diff
@@ -73,6 +73,10 @@ class Question(BaseModel):
     answerFinalizedTimestamp: t.Optional[DatetimeUTC] = None
     currentAnswer: t.Optional[str] = None
 
+    @property
+    def questionId(self) -> HexBytes:
```
Comment (evangriffiths) [inline prediction_market_agent_tooling/markets/omen/data_models.py:77]: the convention we use is snake case for properties
```suggestion
    def question_id(self) -> HexBytes:
```

**Review by evangriffiths — APPROVED** [review]
one nit, but otherwise lgtm

**Comment by kongzii** [comment]
I'll merge these local chain test failures are unrelated to this PR changes

**Pull request merged**